### ISSUE TITLE
allow save for current configuration

### ISSUE THREAD

**Issue by lironalm** [reef-pi/reef-pi#708]
allow save and load config file for configuration backup and restore after crash (SD, user error)

**Comment by ranjib** [comment]
the /var/lib/reef-pi/reef-pi.db file is enough to recreate a reef-pi setup along with a given reef-pi version. I am not sure if this is enough, or we should provide a way for users to download and upload this file from the UI

**Comment by zekth** [comment]
Yes it can be good. But we need a fallbafk instead of a corrupted database. Not a simple gile replacement. Thought @ranjib?

I can do all the frontend stuff for this.

**Comment by kevinmic** [comment]
I just setup a reef-pi today so hopefully its ok if I comment here. 

It is really handy to be able to export/import the current configuration so you can easily get back and running if you have your pi crash.

I have used another open source pi software called sprinkler-pi and it has an export/import configuration feature I find really convenient. It allows for both downloading the configuration JSON or emailing it.

**Comment by zekth** [comment]
> I have used another open source pi software called sprinkler-pi and it has an export/import configuration feature I find really convenient. It allows for both downloading the configuration JSON or emailing it.

ATM all the data is stored in a SQLite db. So serializing it to json and deserializing it to use it have no value. But as i said before we can get the db file.

**Comment by kevinmic** [comment]
If you are looking for a quick and dirty solution then exporting the Blaze DB file would work just fine and serve immediate needs. The biggest downside is it won't be human readable.

**Comment by kevinmic** [comment]
One other option I was thinking about is you technically already have access to the data in JSON format. All you would have to do is hit each API endpoint that you want data from and export that.  Importing the data would essentially be the reverse of this.  It could be a purely UI feature, no reason to change the backend.
